File: 2020S1.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int n;
    cin >> n;

    vector<pair<int,int>> obs(n);

    for (int i = 0; i < n; i++) {
        cin >> obs[i].first >> obs[i].second;
    }

    sort(obs.begin(), obs.end()); // sort by time

    double max_speed = 0.0;

    for (int i = 0; i < n; i++) {
        for (int j = i + 1; j < n; j++) {
            double dt = abs(obs[j].first - obs[i].first);
            double dx = abs(obs[j].second - obs[i].second);
            double speed = dx / dt;
            if (speed > max_speed) {
                max_speed = speed;
            }
        }
    }

    cout << fixed << setprecision(6) << max_speed << endl;
}

File: 2020S1.py
n = int(input())
obs = []

for _ in range(n):
    t, x = map(int, input().split())
    obs.append((t, x))

# Sort by time
obs.sort()

max_speed = 0.0

for i in range(n):
    for j in range(i + 1, n):
        dt = abs(obs[j][0] - obs[i][0])
        dx = abs(obs[j][1] - obs[i][1])
        speed = dx / dt
        if speed > max_speed:
            max_speed = speed

print(max_speed)
